### DISCUSSION TITLE
The keyboard.matrix_positions setting in keyboard config yaml

### DISCUSSION THREAD

**Discussion by fohrloop** [dariogoetz/keyboard_layout_optimizer#80] [Q&A] [answered]
I see the default for 36 keys is (`config/keyboard/chocolad.yml`):

```yml
  matrix_positions:
    - [[2,1], [3,1], [4,1], [5,1], [6,1],                  [13,1], [14,1], [15,1], [16,1], [17,1]]
    - [[2,2], [3,2], [4,2], [5,2], [6,2],                  [13,2], [14,2], [15,2], [16,2], [17,2]]
    - [[2,3], [3,3], [4,3], [5,3], [6,3],                  [13,3], [14,3], [15,3], [16,3], [17,3]]
    - [                     [5,4], [6,4], [7,4],   [12,4], [13,4], [14,4]                        ]
```

If I use for example:

```yml
  matrix_positions:
    - [[5,3], [4,3], [3,3], [2,3], [1,3],                  [11,3], [12,3], [13,3], [14,3], [15,3]]
    - [[5,4], [4,4], [3,4], [2,4], [1,4],                  [11,4], [12,4], [13,4], [14,4], [15,4]]
    - [[5,5], [4,5], [3,5], [2,5], [1,5],                  [11,5], [12,5], [13,5], [14,5], [15,5]]
    -  [              [24,10], [25,10], [26,10],   [36,10], [35,10], [34,10]                     ]
 ```

am I going to face problems? I.e, are these simply identifiers for rows or are these used for some calculations? Note that there's
- Column numbering running from right to left (1 to 5) on the left side of the keyboard.
- Thumb row with number 10
- Left thumb columns: 24, 25, 26
- Right thumb columns: 36, 35, 34 (again, numbers running to "opposite" direction)

**Comment by dariogoetz** [comment]
I think, it would work just fine. But I don't know by heart. Maybe just try out if the evaluation yields different results.

**Reply by dariogoetz** [reply, marked as answer]
PS: the local differences are relevant. E.g. How many rows does an SFB cross. Or are the columns two apart, but the fingers are next to each other (indicating a lateral stretch). For that, the actual numbers are relevant. In your example, the local numbers are fine, though.
